test/functional/vimscript/system_spec.lua

```diff
@@ -660,4 +660,33 @@ describe('shell :!', function()
         1]])
     end
   end)
+
+  it(':{range}! without redirecting to buffer', function()
+    local screen = Screen.new(500, 10)
+    screen:attach()
+    insert([[
+      3
+      1
+      4
+      2]])
+    feed(':4verbose %w !sort<cr>')
+    if is_os('win') then
+      screen:expect{
+        any=[[Executing command: .?sort %< .*]]
+      }
+    else
+      screen:expect{
+        any=[[Executing command: .?%(sort%) %< .*]]
+
+      }
+    end
+    feed('<CR>')
+    helpers.set_shell_powershell(true)
+    feed(':4verbose %w !sort<cr>')
+    screen:expect{
+      any=[[Executing command: .?& { Get%-Content .* | & sort }]]
+    }
+    feed('<CR>')
+    helpers.expect_exit(command, 'qall!')
```

Comment:
To be clear: this is a [questionable workaround](https://github.com/neovim/neovim/pull/20530/files#r1068109057). 

Ideally we would root-cause what's going on here. But it's off-topic for the fix in this PR, so we don't want to block on this.